src/panels/Scene/SceneTree/sortingUtil.ts

```diff
@@ -0,0 +1,169 @@
+import { TreeNodeData } from '@mantine/core';
+
+import { Properties } from '@/types/types';
+import { guiOrderingNumber } from '@/util/propertyTreeHelpers';
+
+import { GroupPrefixKey, isGroupNode } from './treeUtil';
+
+interface TreeSortingInfoEntry {
+  type: 'propertyOwner' | 'group';
+  payload: string;
+  name: string;
+  guiOrder: number | undefined;
+}
+
+interface TreeSortingInfo {
+  [key: string]: TreeSortingInfoEntry;
+}
+
+export function createTreeSortingInformation(
+  treeData: TreeNodeData[],
+  properties: Properties
+) {
+  const result: TreeSortingInfo = {};
+  treeData.forEach((node) => {
+    function addNode(node: TreeNodeData) {
+      if (isGroupNode(node)) {
+        const groupPath = node.value.replace(GroupPrefixKey, '');
+        result[node.value] = {
+          type: 'group',
+          payload: groupPath,
+          name: node.label as string,
+          guiOrder: undefined
+        };
+      } else {
+        // Property owner
+        result[node.value] = {
+          type: 'propertyOwner',
+          payload: node.value,
+          name: node.label as string,
+          guiOrder: guiOrderingNumber(properties, node.value)
+        };
+      }
+
+      if (node.children) {
+        node.children.forEach(addNode);
+      }
+    }
+    addNode(node);
+  });
+
+  return result;
+}
+
+// Sort a list of items in the scene menu tree. This is a bit complicated, since there are
```

Comment:
What comment style to use for functions? Should this be a doxygen-style comment?